### DISCUSSION TITLE
ListItemIndent for tab should use tab

### DISCUSSION THREAD

**Discussion by kangruixiang** [remarkjs/remark#527] [General] [answered]
Currently the listItemIndent option with "tab" uses 4 spaces. I feel it should use /t for tab as this would be more consistent with the option name and would let the individual editor choose how many spaces or tab to use. Thank you.

**Comment by wooorm** [comment, marked as answer]
Unfortunately, actual tab characters don’t work well with markdown. Markdown is a whitespace-sensitive language which uses a hardcoded tab size of 4 characters for tab characters.

My tab size is configured to two spaces, so if I’d use tabs:

```markdown
1.  a

  a
```

The above would look weird: it would look as if there’s a list item followed by a slightly indented paragraph, but is actually properly indented for markdown.
If I’d try and “fix” it by aligning them

```markdown
1.  a

    a
```

I’d actually get a paragraph and a code block in a list item.

---

The name of the option refers to that tab size of 4 spaces. While I understand that it can be confusing, adding actual tab characters doesn’t make sense to me. Feel free to suggest better names though.

**Reply by kangruixiang** [reply]
Gotcha, thank you so much for the explanation.
